interview-questions.md

```diff
@@ -8,36 +8,71 @@ Then, research the question to expand on your answer. Even if you feel you have
+
+class CreateCohorts < ActiveRecord::Migration
+  def change
+    create_table : Cohort do |t|
+      t.string :students
+      t.timestamps
+    end
+  end
+end
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: You can use as many parameters as you want. Therefore, any of the seven RESTful routes can use parameters. 
 
-Researched answer:
+Researched answer: They all use params.
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: rails g controller, rails g model, rails g resoursce
+
+Researched answer: rails g controller creates all files associated with a paticular controller that coordinates interaction b/n the user, the view and model.
+
+$rails g controller creates files through generate coomand that let us make changes to a database.
+
+$rails g model is directly tied to database generate new model which let us add database records. 
 
-Researched answer:
+$rails g resource command will a model that has its columns and data types a controller with its view folder and all the required RESTful routes. 
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
+
 action: "GET" location: /students
 
+"index" controller. It list all sudents in database.
+
 action: "POST" location: /students
 
+"create" controller. It addd the value of students in to a database.
+
 action: "GET" location: /students/new
 
+"new" controller. It returns an HTML FORM for a new student to be inserted in to the database.
+
 action: "GET" location: /students/2
 
+"show" controller. It gives us a return of a specific studentnt2 from the database.
+
 action: "GET" location: /students/2/edit
 
+"edit" controller. It returns an HTML FORM for updating a student 2 in the database.
+
 action: "PATCH" location: /students/2
 
+"update" controller. It allows us to overwrite the value of an existing student 2.
+
 action: "DELETE" location: /students/2
 
+"destroy" controller. It removes (not accessable anymore) the database student2 permanently. 
+
+
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
+??????
```

Comment:
The idea of this assignment was to write 10 features that the user can experience by using an application. So think of ten useful features that an application can provide for a user. These will be written from the user's perspective.
`As a user, I can change the color of my task reminder.`